File: app/src/main/java/com/example/apiproject/java/ui/MainActivity.java
package com.example.apiproject.java.ui;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

import com.example.apiproject.R;
import com.example.apiproject.java.data.model.FlowerResponse;
import com.example.apiproject.java.data.repository.FlowerRepository;
import com.example.apiproject.java.data.repository.FlowerRepositoryImpl;

import java.util.List;

public class MainActivity extends AppCompatActivity implements MainContract.View{

    private static final String TAG = "MainActivity";
    private MainContract.Persenter presenter;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initPresenter();

    }

    private void initPresenter() {
        FlowerRepository repository= new FlowerRepositoryImpl();
        presenter= new MainPresenter(this, repository);
        presenter.handle();
    }

    @Override
    public void updateView(List<FlowerResponse.Hits> hits) {
        Log.d(TAG, "updateView: view da duoc cap nhat");
        Log.d(TAG, "updateView: "+hits.size());
    }
}

File: app/src/main/java/com/example/init/ui/MainActivity.kt
package com.example.init.ui

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.ViewModelProvider
import com.example.init.R
import com.example.init.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    lateinit var binding: ActivityMainBinding
    lateinit var viewModel: MainViewModel
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        viewModel = ViewModelProvider(this).get(MainViewModel::class.java)
        binding.viewModel = viewModel
        binding.lifecycleOwner = this
    }
}